### PR TITLE
Add variable probability weights in the Custom Sampler

### PR DESCRIPTION
**Pull request by wangcj05**
--------
Pull Request Description
--------
##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

**Comment by wangcj05** [comment]
```
**** SUMMARY ****
*********************************
**** THE FOLLOWING TESTS CHANGED OUTCOMES ****
**** Test framework.external_xml | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework.getpot_with_externalXML | has changed outcome from <SKIPPED> to <FAILED> !!!!
**** Test framework/CodeInterfaceTests.MAAP5interfaceForwardSampling | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/Distributions.categorical | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/BasicStatistics.timeDepOutCsv | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/BasicStatistics.timeDependent | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/InterfacedPostProcessor.RiskMeasuresDiscreteMultipleIE | has changed outcome from <FAILED> to <OK> !!!!
*****************
Number of success first file:
345
Number of success second file:
351
NUMBER OF SUCCESSES INCREASED: +6
TOTAL TIME OF TESTS DECREASED       : reference_time(672.1) > new_time(651.3)
AVERAGE TIMES FOR SUCCESS TESTS ARE : reference_time(1.94811594203) v.s. new_time(1.85555555556)
*********************************
```

**Comment by wangcj05** [comment]
This PR is ready for reviewing?

File: framework/Samplers/CustomSampler.py [hunk 1]
```diff
@@ -166,12 +192,10 @@ def localGenerateInput(self,model,myInput):
...
-      self.inputInfo['ProbabilityWeight'] = self.infoFromCustom['ProbabilityWeight'][self.counter-1]
-    else:
-      self.inputInfo['ProbabilityWeight'] = 1.0
-
+      for subVar in var.split(','):
+        subVar = subVar.strip()
+        self.values[subVar] = self.pointsToSample[subVar][self.counter-1]
+        self.inputInfo['ProbabilityWeight-' + subVar] = self.infoFromCustom['ProbabilityWeight-' + subVar][self.counter-1]
+    self.inputInfo['PointProbability'] = self.infoFromCustom['PointProbability'][self.counter-1]
+    self.inputInfo['ProbabilityWeight'] = self.infoFromCustom['ProbabilityWeight'][self.counter-1]
```
Comment (PaulTalbot-INL) [inline framework/Samplers/CustomSampler.py]: I don't see any comments in all of this.  Can we put a comment before each significant set of actions that explains what is happening in human-readable form?  Things like "check requested inputs and outputs are available" and "construct missing probabilities", etc.

**Comment by wangcj05** [comment]
```
[wangc@falcon2 raven]$ python compare_tests.py reference_test_results.txt new_development_test_results.txt
**** SUMMARY ****
*********************************
**** THE FOLLOWING TESTS CHANGED OUTCOMES ****
**** Test framework.external_xml | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework.getpot_with_externalXML | has changed outcome from <SKIPPED> to <FAILED> !!!!
**** Test framework/CodeInterfaceTests.MAAP5interfaceForwardSampling | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/Distributions.categorical | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/BasicStatistics.timeDepOutCsv | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/BasicStatistics.timeDependent | has changed outcome from <FAILED> to <OK> !!!!
**** Test framework/PostProcessors/InterfacedPostProcessor.RiskMeasuresDiscreteMultipleIE | has changed outcome from <FAILED> to <OK> !!!!
**** Test plugins/ExamplePlugin.TestPluginExample | has changed outcome from <FAILED> to <OK> !!!!
**** Test plugins/ExamplePlugin/tests.TestPluginExample | has changed outcome from <FAILED> to <OK> !!!!
*****************
Number of success first file:
373
Number of success second file:
381
NUMBER OF SUCCESSES INCREASED: +8
TOTAL TIME OF TESTS INCREASED       : reference_time(447.8) < new_time(454.2)
AVERAGE TIMES FOR SUCCESS TESTS ARE : reference_time(1.20053619303) v.s. new_time(1.19212598425)
*********************************
```

**Pull request merged**